web/src/message_edit.js

```diff
@@ -534,6 +550,7 @@ function edit_message($row, raw_content) {
                 // since it isn't clear what to do with the half-finished edit. It's nice to keep
                 // the half-finished edit around so that they can copy-paste it, but we don't want
                 // people to think "Save" will save the half-finished edit.
+                $message_edit_saving_container.hide();
```

Comment:
Hmm, why do you hide this here? It seems like that operation hasn't finished...

Reply:
So this code path gets executed when the time limit for editing the message has exceeded. When the time limit gets executed, I felt it is better to hide the "Saving" button, while we show the "Save" as disabled, just to continue showing as it was before.

Reply:
So the server does have a ~20-second grace period, so I think in most cases where this occurs, the operation will succeed. So this might not be the right strategy?

Reply:
yup, I guess you're right -- the server adds up to 10s time if the edited message is saved just before completion time. I updated to remove that line.